src/shared/lib/stringUtils.ts

```diff
@@ -30,15 +30,28 @@ export function joinOxfordComma(strs: string[]): string {
 }
 
 /**
- * Split the input string non-letter characters (\P{L}) and see if any word matches the inputted query.
+ * Normalize a string by removing accent marks (diacritics) for accent-insensitive comparison.
+ * Uses NFD normalization followed by removal of combining diacritical marks.
+ */
+export function normalizeAccents(str: string): string {
+  return str.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
+}
+
+/**
+ * Split the input string on non-alphanumeric characters and see if any word matches the inputted query.
+ * Splits on characters that are neither letters (\p{L}) nor numbers (\p{N}), so alphanumeric codes
  * Sometimes inputted queries may contain whitespace -- for that cause we check if the general
  * input also matches.
+ * Both the string and query are normalized to be accent-insensitive and case-insensitive.
  */
-export function anyWordStartsWith(str: string, lowercaseQuery: string) {
+export function anyWordStartsWith(str: string, query: string) {
   const strLowercase = str.toLowerCase();
+  const queryLowercase = query.toLowerCase();
+  const normalizedStr = normalizeAccents(strLowercase);
+  const normalizedQuery = normalizeAccents(queryLowercase);
```

Comment:
Sometimes its good to split variable declarations for clarity. Here, you can combine them.

```suggestion
  const normalizedStr = normalizeAccents(str.toLowerCase());
  const normalizedQuery = normalizeAccents(query.toLowerCase());
```